htdocs/inc/functions.inc.php

```diff
@@ -1776,3 +1764,47 @@ function collect(array $data): ArrayCollection
 {
     return new ArrayCollection($data);
 }
+
+if (! function_exists('blank')) {
```

Comment:
Should really avoid copying code from other projects. If we use code from somewhere we should require the lib via Composer

Reply:
Any other recommendations for these kind of practical helper functions? other libraries?

Illuminate/support would pull in some more dependencies, question is if this is desired:

![image](https://user-images.githubusercontent.com/1665428/123974465-e7256a00-d9bc-11eb-8009-44e3e376751a.png)

Reply:
In general adding dependencies, even dependencies with other dependencies is not a bad thing. In this particular case I am not sure it is necessary. If I see correct then it is only used for the message subject. I think here we could easily just check where we create the message object whether subject is `null` or empty string and if either is the case set it to null. Then all we have to check in the application is whether the subject is null.

Reply:
Currently it is used only there, but the filled() and blank() methods are useful in many places in the code

ok, in theory there should be a proper input validation system which takes care of that, but this is still a way to go....

Reply:
right, should maybe discuss how such input validation would look like. Is this just a set of function, do we use something like symfony forms/validator or is this something else entire?

Reply:
Would be great to have a standardized and established approach. I dont know symfony forms validatior, but can have a look at it

Reply:
agree to having a standardized approach, never preference on what to use as long as there is helpful documentation around